docs/how-to/search-job.rst

```diff
@@ -0,0 +1,55 @@
+Search Jobs
+=============
+
+You can search for jobs based on the following criteria:
+   * tags
+   * state
+
+
+Searching by Tags
+------------------
+
+Tags can make it easier to find related jobs later. For instance, tools like
+spread have a need to find currently running jobs so that they can be
+cancelled once they are done using those devices for their testing.
+
+To add tags to a job submission, add a section like this with one or more tags
+in your job yaml:
+
+.. code-block:: yaml
+
+      tags:
+        - myproject
+        - client
+
+Testflinger doesn't use any of this information, but it makes it easier for
+you to search for those jobs later.
+
+The search API allows you to search for jobs based on tags and state. To
+search for a jobs by tag, you can provide one or more tags in the query string:
+
+.. code-block:: console
+
+      $ curl 'http://localhost:8000/v1/job/search?tags=foo&tags=bar'
+
+By default, the search API will return jobs that match any of the tags. To
+require that all tags match, you can provide the "match" query parameter with
+the value "all":
+
+.. code-block:: console
+
+      $ curl 'http://localhost:8000/v1/job/search?tags=foo&tags=bar&match=all'
+
+Searching by Job State
+-----------------------
+
+By default, the search API will only jobs that are not already marked as
+cancelled or completed. To specify searching for jobs in a specific state, you
+can provide the "state" query parameter with one of the
```

Comment:
Following the principle of least astonishment, I would've expected the default to return all jobs regardless of state given that I didn't specify state

Reply:
> a really long search string for the common case. ex ?tags=foo&state=waiting&state=test&state=provision&state=allocated...

I see the issue. One idea then is to just expose through the api docs/schema that there is this default value for the state field. I was wondering whether we could use the property `dump_default` from `Field` model. But I'm not sure if it would work for lists. But if it does, then we could possibly specify the default through the schema and avoid handling it in code

Reply:
Ok, check out the latest revision. I added a fake state called "active". If you specify this state, then it only considers jobs that are not "cancelled" or "completed".  Otherwise it considers all states unless you specify states to filter on.

Reply:
I love it 👍